integration/src/main/java/scripts/testscripts/SnapshotDatasetOperations.java

```diff
@@ -0,0 +1,165 @@
+  private static final Logger log = LoggerFactory.getLogger(SnapshotDatasetOperations.class);
+  private static final String TEST_DATASET_NAME = "CatalogTestDataset";
+
+  // TDR APIs
+  private SnapshotsApi snapshotsApi;
+  private UUID snapshotId;
+
+  // Catalog APis
+  private UUID datasetId;
+  private DatasetsApi datasetsApi;
+
+  @Override
+  public void setup(List<TestUserSpecification> testUsers) throws Exception {
+    DataRepoClient datarepoClient = new DataRepoClient(server, testUsers.get(0));
+    var datasetsApi = new bio.terra.datarepo.api.DatasetsApi(datarepoClient);
+    var dataset =
+        datasetsApi
+            .enumerateDatasets(null, null, null, null, TEST_DATASET_NAME, null)
+            .getItems()
+            .get(0);
+
+    snapshotsApi = new SnapshotsApi(datarepoClient);
+    var request =
+        new SnapshotRequestModel()
+            .name("catalog_integration_test_" + System.currentTimeMillis())
+            .description("catalog test snapshot")
+            .profileId(dataset.getDefaultProfileId())
+            .addContentsItem(
+                new SnapshotRequestContentsModel()
+                    .datasetName(TEST_DATASET_NAME)
+                    .mode(SnapshotRequestContentsModel.ModeEnum.BYFULLVIEW));
+    var createJobId = snapshotsApi.createSnapshot(request).getId();
+
+    var jobApi = new JobsApi(datarepoClient);
+    JobModel job;
+    do {
+      job = jobApi.retrieveJob(createJobId);
+    } while (job.getJobStatus() == JobModel.JobStatusEnum.RUNNING);
+
+    if (job.getJobStatus() != JobModel.JobStatusEnum.SUCCEEDED) {
+      throw new RuntimeException("Create snapshot failed: " + job);
+    }
+    @SuppressWarnings("unchecked")
+    Map<Object, Object> result = (Map<Object, Object>) jobApi.retrieveJobResult(createJobId);
+    snapshotId = UUID.fromString((String) result.get("id"));
+    log.info("created snapshot " + snapshotId);
+  }
+
+  private static final String METADATA_1 = """
+      {"name": "test"}""";
+
+  private static final String METADATA_2 = """
+      {"name": "test2"}""";
+
+  @Override
+  public void userJourney(TestUserSpecification testUser) throws Exception {
+    var client = new CatalogClient(server, testUser);
+    datasetsApi = new DatasetsApi(client);
+
+    // Given a snapshot, create a catalog entry.
+    var request =
+        new CreateDatasetRequest()
+            .catalogEntry(METADATA_1)
+            .storageSourceId(snapshotId.toString())
+            .storageSystem(StorageSystem.TDR);
+    datasetId = datasetsApi.createDataset(request).getId();
+    assertThat(client.getStatusCode(), is(HttpStatusCodes.STATUS_CODE_OK));
+    assertThat(datasetId, notNullValue());
+    log.info("created dataset " + datasetId);
+
+    // Retrieve the entry
+    assertThat(datasetsApi.getDataset(datasetId), is(METADATA_1));
+    assertThat(client.getStatusCode(), is(HttpStatusCodes.STATUS_CODE_OK));
+
+    // Retrieve all datasets
+    var datasets = datasetsApi.listDatasets();
```

Comment:
Are these methods actually making a new rest call? It seems a bit like they're making internal calls to the services?

Reply:
Yes, this is making an API call to a server that's running in a separate process. The `ApiClient` class is the swagger generated client code which is a wrapper around `javax.ws.rs.client.Client`. One way to see this in action is to run the tests locally, where you can see the HTTP requests in the locally running server logs.